programs/perpetuals/src/state/stake.rs

```diff
@@ -0,0 +1,173 @@
+    // incentivizing claiming for other participants with the use of this bounty.
+    // The goal of this is to prevent users to loose their rewards, as the StakingRound will be dropped after that. This
+    // can be seen as a sort of "auto restaking".
+    // The bounty amount depend of the age of the Stake account, the closer to `MAX_AGE_SECONDS` the higher.
+    // Bounty starts at 90%-95% for 0.1%, then gets bumped from 95%-100% to 5%
+    pub fn get_claim_stake_caller_reward_token_amounts(
+        &self,
+        reward_token_amount: u64,
+        current_time: i64,
+    ) -> Result<u64> {
+        let bounty_stage = self.bounty_stage(current_time)?;
+        match bounty_stage {
+            BountyStage::NoReward => Ok(0),
+            BountyStage::StageOne => Ok(max(
+                1,
+                math::checked_decimal_mul(
+                    reward_token_amount,
+                    1,
+                    bounty_stage.reward_bps(),
+                    -(Perpetuals::BPS_DECIMALS as i32),
+                    0,
+                )?,
+            )),
+            BountyStage::StageTwo => Ok(max(
+                1,
+                math::checked_as_u64(math::checked_div(
+                    math::checked_mul(
+                        reward_token_amount as u128,
+                        bounty_stage.reward_bps() as u128,
+                    )?,
+                    Perpetuals::BPS_POWER,
+                )?)?,
+            )),
+        }
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+
+    fn get_fixture_stake(stake_time: i64) -> Stake {
+        Stake {
+            amount: 0,
+            bump: 255,
+            stake_time,
+        }
+    }
+
+    #[test]
+    fn test_get_claim_stake_caller_reward_token_amounts() {
+        let reward_token_amount = 100; // native units
+
+        // out of the bounty period
+        let time = 69420;
+        let stake = get_fixture_stake(time);
+        let current_time = time + 0;
+        let bounty_amount = stake
+            .get_claim_stake_caller_reward_token_amounts(reward_token_amount, current_time)
+            .unwrap();
+        assert_eq!(bounty_amount, 0);
+
+        // in of the bounty period phase one
+        let time = 69420;
+        let stake = get_fixture_stake(time);
+        let current_time = time + 28386000; //90% of a year
+        let bounty_amount_phase_one = stake
+            .get_claim_stake_caller_reward_token_amounts(reward_token_amount, current_time)
+            .unwrap();
+        assert_ne!(bounty_amount_phase_one, 0);
+
+        // in of the bounty period phase two
+        let time = 69420;
+        let stake = get_fixture_stake(time);
+        let current_time = time + 29979079; // 95% of a year
+        let bounty_amount_phase_two = stake
+            .get_claim_stake_caller_reward_token_amounts(reward_token_amount, current_time)
+            .unwrap();
+        assert!(bounty_amount_phase_one < bounty_amount_phase_two);
```

Comment:
nit: I like using `_` as separator for numbers

```suggestion
        let time = 69_420;
        let stake = get_fixture_stake(time);
        let current_time = time + 0;
        let bounty_amount = stake
            .get_claim_stake_caller_reward_token_amounts(reward_token_amount, current_time)
            .unwrap();
        assert_eq!(bounty_amount, 0);

        // in of the bounty period phase one
        let time = 69_420;
        let stake = get_fixture_stake(time);
        let current_time = time + 28_386_000; //90% of a year
        let bounty_amount_phase_one = stake
            .get_claim_stake_caller_reward_token_amounts(reward_token_amount, current_time)
            .unwrap();
        assert_ne!(bounty_amount_phase_one, 0);

        // in of the bounty period phase two
        let time = 69_420;
        let stake = get_fixture_stake(time);
        let current_time = time + 29_979_079; // 95% of a year
        let bounty_amount_phase_two = stake
            .get_claim_stake_caller_reward_token_amounts(reward_token_amount, current_time)
            .unwrap();
        assert!(bounty_amount_phase_one < bounty_amount_phase_two);
```